src/components/InstanceProvider.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState, FC, useEffect } from 'react';
+import { InstanceContext } from 'components/InstanceContext';
+import { GlobalSettings, GrafanaInstances } from 'types';
+import { getDataSourceSrv } from '@grafana/runtime';
+import { SMDataSource } from 'datasource/DataSource';
+import { Spinner } from '@grafana/ui';
+import { AppPluginMeta } from '@grafana/data';
+import { UnprovisionedSetup } from 'components/UnprovisionedSetup';
+
+async function fetchDatasources(metricInstanceName: string, logsInstanceName: string): Promise<GrafanaInstances> {
+  const dataSourceSrv = getDataSourceSrv();
+  const smApi = (await dataSourceSrv.get('Synthetic Monitoring').catch(e => undefined)) as SMDataSource | undefined;
+  const metrics = await dataSourceSrv.get(metricInstanceName).catch(e => undefined);
+  const logs = await dataSourceSrv.get(logsInstanceName).catch(e => undefined);
```

Comment:
When a user logs in for the first time (especially when running on prem) it's possible/likely that none of the datasources the plugin depends on exist. So this is supposed to be 'get the datasources if they exist, if they don't we're going to have to kick the user into a different flow' instead of guaranteeing their existence